data/txHandler.go

```diff
@@ -529,6 +558,24 @@ func (handler *TxHandler) processIncomingTxn(rawmsg network.IncomingMessage) net
 	dec := protocol.NewMsgpDecoderBytes(rawmsg.Data)
 	ntx := 0
 	consumed := 0
+
+	var err error
+	var capguard *util.ErlCapacityGuard
+	if handler.erl != nil {
+		// consume a capacity unit
+		capguard, err = handler.erl.ConsumeCapacity(rawmsg.Sender.(util.ErlClient))
```

Comment:
What makes you sure this will always return a valid pointer? `rawmsg.Sender.(util.ErlClient)`

Reply:
`rawmsg.Sender` is a `Peer` interface, and my understanding is that the only in-use implementation of a Peer is the `wsPeer`.

`wsPeer` satisfies `ErlClient` by implementing `OnClose`.

Since `Peer` is just an empty interface, it makes it hard to say I'm "sure" it will always be a valid pointer.

This looks like it might be a go pattern I have yet to learn(?)

Reply:
> Since Peer is just an empty interface, it makes it hard to say I'm "sure" it will always be a valid pointer.

Yes, and we should not be adding more into this. 

It is really not worth taking a chance here.
```suggestion
		capguard, err = handler.erl.ConsumeCapacity(rawmsg.Sender)
```
Then, inside consume capacity:
```
ConsumeCapacity(p Peer) {
	erlclient, ok := p.(util.ErlClient)
	if !ok {
		return nil, error("unsupported peer")
	}
...
}